src/optimizer/pushdown/pushdown_projection.cpp

```diff
@@ -6,6 +6,23 @@
 
 namespace duckdb {
 
+static bool HasSideEffects(LogicalProjection &proj, const unique_ptr<Expression> &expr) {
+	if (expr->type == ExpressionType::BOUND_COLUMN_REF) {
+		auto &colref = expr->Cast<BoundColumnRefExpression>();
+		D_ASSERT(colref.binding.table_index == proj.table_index);
+		D_ASSERT(colref.binding.column_index < proj.expressions.size());
+		D_ASSERT(colref.depth == 0);
```

Comment:
Can you add a test where all of this plays out within a subquery? essentially nesting it:
```sql
select * from (select rnd from (select random()) as t(rnd) where rnd < 0.5);
```

Reply:
Add it. Thanks!